File: gcj2012/speaking_in_tongues.py
#!/usr/bin/python2.5
#-*- coding: utf-8 -*-

# Pychecker options
__pychecker__ = 'no-callinit no-classattr'

# External imports
import string
import sys

# Internal imports (if any)

def getline():
  return sys.stdin.next().rstrip()

transtable = string.maketrans(
  'abcdefghijklmnopqrstuvwxyz',
  'yhesocvxduiglbkrztnwjpfmaq')
def challenge():
  print getline().translate(transtable)

# Main entry point
if __name__ == '__main__':
  testcases = int(getline())

  for testcase in xrange(1, testcases + 1):
    print 'Case #%d:' % (testcase, ),
    challenge()

File: gcj2012/speaking_in_tongues.cpp
#include <iostream>
#include <string>

char transtable[] = "yhesocvxduiglbkrztnwjpfmaq";
void challenge()
{
  std::string input, output;
  std::getline(std::cin, input);

  for (std::string::const_iterator c = input.begin(); c != input.end(); ++c)
  {
    if (*c < 'a' || *c > 'z')
      output += *c;
    else
      output += transtable[*c - 'a'];
  }

  std::cout << output << std::endl;
}

int main(int argc, char** argv)
{
  unsigned int testcases;
  std::cin >> testcases;

  {
    // Discard first line
    std::string firstline;
    std::getline(std::cin, firstline);
  }

  for (unsigned int testcase = 1; testcase <= testcases; ++testcase)
  {
    std::cout << "Case #" << testcase << ": ";
    challenge();
  }

  return 0;
}

